### DISCUSSION TITLE
移动端的这种流式布局应该怎么弄，如图所示

### DISCUSSION THREAD

**Discussion by YSW33** [irihitech/Ursa.Avalonia#204] [Q&A]
![image](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/7d2c2a64-e3ac-49a2-9a61-e707c7e4bfbc)
每个单元按钮数量不固定。

**Comment by rabbitism** [comment]
uniform grid？

**Comment by YSW33** [comment]
> uniform grid？

我现在就是用ItemsControl+GRID实现的，但是要提前计算好控件宽和高。
![image](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/918af017-ffa5-4c88-9386-67043ba0c5a5)

**Comment by YSW33** [comment]
![8baf49ff7e3eed222461be5ab52cdba](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/1e9b0808-1446-4dc2-9bc1-510b2575c5ad)

**Comment by rabbitism** [comment]
No，我是说UniformGrid，只要设置列数就好了

**Comment by YSW33** [comment]
> No，我是说UniformGrid，只要设置列数就好了

我试试。

**Comment by YSW33** [comment]
> No，我是说UniformGrid，只要设置列数就好了

1. UniformGrid 的Item 该如何设置DataTemplate?
2. 我看源代码是Panel，也没有ItemsSource，只能通过Children代码来添加item。

**Comment by rabbitism** [comment]
You can check this sample. 

https://github.com/irihitech/Avalonia-Tutorials/tree/main/Tutorials/Use_ItemsPanel

**Reply by YSW33** [reply]
THANKS

**Reply by YSW33** [reply]
这个类似，但是我是ItemsControl 嵌套 ItemsControl的效果，这个该如何处理？

				<ItemsControl.ItemTemplate>
					<DataTemplate>
						<Border CornerRadius="5" BorderThickness="1" BorderBrush="Black" Margin="0,0,0,5">
							<control:ItemGroupControl/>
						</Border>
					</DataTemplate>
				</ItemsControl.ItemTemplate>
			</ItemsControl >

**Comment by YSW33** [comment]
![56c62e63b2fd03be83f4bec19ab1f3f](https://github.com/irihitech/Ursa.Avalonia/assets/141573398/4845bd5b-e959-41cd-9093-115ed10a4de1)
这是我现在实现的效果
ItemsControl的DataTemplate使用自定义ItemGroupControl，ItemGroupControl的控件大小是加载时候重新计算，如果用ItemsControl嵌套ItemsControl，控件大小就会便变成默认大小，所有图标挤在一起。
